app/models/course.rb

```diff
@@ -149,6 +149,27 @@ def application_forms
       .distinct
   end
 
+  def self.find_courses_in_previous_cycle(provider)
```

Comment:
These `find_` methods should be ActiveRecord scopes, or if they are too complex for this we should define a queery class. 
I don't think the method names reflect what the methods do as there are joins and includes here, it's not obvious what these are for.
Also prefixing methods with `find_` can be ambiguous in our codebase because we tend to reference the Find application this way.

Reply:
Thanks, I've removed these find methods and chained scopes in `generate_test_applications_for_provider.rb`